rootfs/scheduler/resources/deployment.py

```diff
@@ -346,3 +346,6 @@ def wait_until_ready(self, namespace, name, **kwargs):
 
             waited += 1
             time.sleep(1)
+        ready, _ = self.are_replicas_ready(namespace, name)
```

Comment:
Can you add a comment and separate it a bit from the for loop.

My only reservation is so far is that we have to wait until the timeout is done, which for some people can be a long time and the connection can be severed, tho at the same time if that is the case our system wouldn't go ahead with any cleanups either so the problem wouldn't be around